### PR TITLE
[api] Adds FUSE based repository support

### PR DESCRIPTION
**Pull request by raymondkhliu**
1. No cloud SDK dependencies
2. No need to download the entire bucket
3. High performance

## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: api/src/test/java/ai/djl/repository/FuseRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        } finally {
+            System.clearProperty("GCSFUSE");
+            System.clearProperty("DJL_CACHE_DIR");
+        }
+    }
+
+    @Test
+    public void testS3Repository() throws IOException {
+        if (System.getProperty("os.name").startsWith("Win")) {
+            throw new SkipException("GcsRepository is not supported on Windows");
```
Comment (frankfliu) [inline api/src/test/java/ai/djl/repository/FuseRepositoryTest.java:58]:
```suggestion
            throw new SkipException("S3Repository is not supported on Windows");
```

**Review by frankfliu — APPROVED** [review]
